src/Main.cpp

```diff
@@ -0,0 +1,177 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+#include <algorithm>
+#include "Location.h"
+#include "DistanceCalculator.h"
+
+std::map<std::string, Location> dataMap;
+
+void readFile() {
+    std::string filePath;
+    std::cout << "Enter the file path: ";
+    std::cin >> filePath;
+    std::ifstream file(filePath); // Replace "input.txt" with the path to your file
+    //std::ifstream file("C:\\exelteam_c\\task6_yaniv\\resources\\data.txt")
+    
+    if (!file.is_open()) {
+        throw std::runtime_error( "Failed to open the file.");
+    }
+
+    std::string line;
+    std::string locationName;
+
+    while (std::getline(file, line)) {
+        locationName = line;
+        if (std::getline(file, line)) {
+            size_t spacePos = line.find(' ');
+            Location location;
+            location.latitude = std::stod(line.substr(0, spacePos));
+            location.longitude = std::stod(line.substr(spacePos + 3));
+            dataMap[locationName] = location;
+        }
+    }
+    file.close();
+
+
+}
+
+
+// Predicate to check if a city is within the given radius from the selected city
+bool isWithinRadius(const std::pair<std::string, Location>& cityData, const Location& selectedCityData,  int radius, 
+                    const DistanceCalculator& distanceCalculator) {
+    return distanceCalculator.calculateDistance(cityData.second, selectedCityData) <= radius;
+}
+
+// Predicate to check if a city is to the north of the selected city
+bool isToNorth(const Location& cityLocation , const Location& selectedCityLocation) {
+    return cityLocation.latitude > selectedCityLocation.latitude;
+}
+
+// Print the details based on the given city, radius, and norm
+void printDetails(const std::string& city, double radius, int norm) {
+    auto iter = dataMap.find(city);
+    if (iter == dataMap.end()) {
+        throw std::runtime_error("ERROR: "+ city+" isn't found in the city list. Please try again.");
+    }
+    Location selectedLocation = iter->second;
+    
+    DistanceCalculator distanceCalculator(norm);
+
```

Comment:
you are missing 1 step in finding the cities in radius, you should find vertical and horizontal rectangles, and then to find the intersection square, and to find a circle inside the square.